server/sonar-server/src/main/java/org/sonar/server/measure/ws/SearchAction.java

```diff
@@ -142,18 +145,26 @@ private SearchRequest setRequest() {
 
     private List<MetricDto> searchMetrics() {
       requireNonNull(request);
-      return dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
+      List<MetricDto> dbMetrics = dbClient.metricDao().selectByKeys(dbSession, request.getMetricKeys());
+      List<String> metricKeys = dbMetrics.stream().map(MetricDto::getKey).collect(Collectors.toList());
+      checkRequest(request.getMetricKeys().size() == dbMetrics.size(), "The following metrics are not found: %s",
+        String.join(", ", difference(request.getMetricKeys(), metricKeys)));
+      return dbMetrics;
```

Comment:
we're not clear about the good strategy about that, fail fast or be tolerant. Should we fail when we delete something that doesn't exist?
In this specific case, it makes more sense to fail. The user has given a metric we don't know, probably a typo or something that the user should be aware of. It's an incorrect query
The same stands for components. If we don't know about the component it means the query is incorrect